### PR TITLE
Fix topK/topKWeighted merge (with non-default parameters)

### PR DESCRIPTION
**Pull request by azat**
Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix topK/topKWeighted merge (with non-default parameters)

### CODE REVIEW

**Review by alexey-milovidov — APPROVED** [review]

File: tests/queries/0_stateless/01356_initialize_aggregation.sql [hunk 1]
```diff
@@ -1,4 +1,5 @@
 SELECT uniqMerge(state) FROM (SELECT initializeAggregation('uniqState', number % 3) AS state FROM system.numbers LIMIT 10000);
 SELECT topKWeightedMerge(10)(state) FROM (SELECT initializeAggregation('topKWeightedState(10)', number, number) AS state FROM system.numbers LIMIT 1000);
 SELECT topKWeightedMerge(10)(state) FROM (SELECT initializeAggregation('topKWeightedState(10)', 1, number) AS state FROM system.numbers LIMIT 1000);
-SELECT topKWeightedMerge(10)(state) FROM (SELECT initializeAggregation('topKWeightedState(10)', number, 1) AS state FROM system.numbers LIMIT 1000);
+-- order in libstdc++ and libc++ may be a little different, that's why we need arraySort()
+SELECT arraySort(topKWeightedMerge(10)(state)) FROM (SELECT initializeAggregation('topKWeightedState(10)', number, 1) AS state FROM system.numbers LIMIT 1000);
```
Comment (azat) [inline tests/queries/0_stateless/01356_initialize_aggregation.sql:5]: The following patch will have the same effect:

```patch
diff --git a/src/Common/SpaceSaving.h b/src/Common/SpaceSaving.h
index 5606334024..0f148568da 100644
--- a/src/Common/SpaceSaving.h
+++ b/src/Common/SpaceSaving.h
@@ -241,7 +241,7 @@ public:
             }
         }
 
-        std::sort(counter_list.begin(), counter_list.end(), [](Counter * l, Counter * r) { return *l > *r; });
+        std::stable_sort(counter_list.begin(), counter_list.end(), [](Counter * l, Counter * r) { return *l > *r; });
 
         if (counter_list.size() > m_capacity)
         {
```

Comment (alexey-milovidov) [reply tests/queries/0_stateless/01356_initialize_aggregation.sql:5]: Ok, but we don't actually need stable sort.

**Comment by alexey-milovidov** [comment]
> test_materialize_mysql_database

@zhang2014 is working on it.

File: tests/queries/0_stateless/01356_initialize_aggregation.sql (hunk 1 above)
Comment (azat) [reply tests/queries/0_stateless/01356_initialize_aggregation.sql:5]: Indeed, that's why arraySort

**Pull request merged**